<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Results</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #232629;
    color: #7a7c7d;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #7a7c7d;  padding-left: 4px; }
div.sourceCode
  { color: #cfcfc2; background-color: #232629; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span { color: #cfcfc2; } /* Normal */
code span.al { color: #95da4c; background-color: #4d1f24; font-weight: bold; } /* Alert */
code span.an { color: #3f8058; } /* Annotation */
code span.at { color: #2980b9; } /* Attribute */
code span.bn { color: #f67400; } /* BaseN */
code span.bu { color: #7f8c8d; } /* BuiltIn */
code span.cf { color: #fdbc4b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #3daee9; } /* Char */
code span.cn { color: #27aeae; font-weight: bold; } /* Constant */
code span.co { color: #7a7c7d; } /* Comment */
code span.cv { color: #7f8c8d; } /* CommentVar */
code span.do { color: #a43340; } /* Documentation */
code span.dt { color: #2980b9; } /* DataType */
code span.dv { color: #f67400; } /* DecVal */
code span.er { color: #da4453; text-decoration: underline; } /* Error */
code span.ex { color: #0099ff; font-weight: bold; } /* Extension */
code span.fl { color: #f67400; } /* Float */
code span.fu { color: #8e44ad; } /* Function */
code span.im { color: #27ae60; } /* Import */
code span.in { color: #c45b00; } /* Information */
code span.kw { color: #cfcfc2; font-weight: bold; } /* Keyword */
code span.op { color: #cfcfc2; } /* Operator */
code span.ot { color: #27ae60; } /* Other */
code span.pp { color: #27ae60; } /* Preprocessor */
code span.re { color: #2980b9; background-color: #153042; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #da4453; } /* SpecialString */
code span.st { color: #f44f4f; } /* String */
code span.va { color: #27aeae; } /* Variable */
code span.vs { color: #da4453; } /* VerbatimString */
code span.wa { color: #da4453; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PredictOutbredCrossVar</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="PreprocessDataFiles.html">Preprocess Data</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Fit MtBRRs
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="fitMtBRRs.html">Run mixed models</a>
    </li>
    <li>
      <a href="getPMVarComps.html">Get VarComps</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Cross-validation study
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="predictCrossVars.html">Predict cross means and (co)variances</a>
    </li>
    <li>
      <a href="observedCrossVars.html">Observed means and (co)variances</a>
    </li>
    <li>
      <a href="predictionAccuracy.html">Prediction accuracy</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Exploratory analyses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="predictUntestedCrosses.html">Predict untested crosses</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Manuscript
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Results.html">Results</a>
    </li>
    <li>
      <a href="Figures.html">Figures</a>
    </li>
    <li>
      <a href="SupplementaryTables.html">Supplementary Tables</a>
    </li>
    <li>
      <a href="SupplementaryFigures.html">Supplementary Figures</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/PredictOutbredCrossVar">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Results</h1>
<h4 class="date">2021-March-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>PredictOutbredCrossVar/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20191123code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20191123)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20191123code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20191123)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdPredictOutbredCrossVartree2ec2ff8186adf2c50e98b23aee4f119f1a86903ftargetblank2ec2ff8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/PredictOutbredCrossVar/tree/2ec2ff8186adf2c50e98b23aee4f119f1a86903f" target="_blank">2ec2ff8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdPredictOutbredCrossVartree2ec2ff8186adf2c50e98b23aee4f119f1a86903ftargetblank2ec2ff8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/PredictOutbredCrossVar/tree/2ec2ff8186adf2c50e98b23aee4f119f1a86903f" target="_blank">2ec2ff8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  Icon
    Untracked:  PredictOutbredCrossVarMS_ResponseToReviews_R1.gdoc
    Untracked:  figure/
    Untracked:  manuscript/
    Untracked:  output/crossPredictions/
    Untracked:  output/gblups_DirectionalDom_parentwise_crossVal_folds.rds
    Untracked:  output/gblups_geneticgroups.rds
    Untracked:  output/gblups_parentwise_crossVal_folds.rds
    Untracked:  output/mtMarkerEffects/

Unstaged changes:
    Modified:   analysis/NGCleadersCall.Rmd
    Modified:   code/fitDirectionalDomMtBRR.R
    Modified:   code/fitmtBRR.R
    Modified:   code/getDirectionalDomGenomicBLUPs.R
    Modified:   code/getDirectionalDomMtCrossMeanPreds.R
    Modified:   code/getDirectionalDomMtCrossVarBVpreds.R
    Modified:   code/getDirectionalDomMtCrossVarTGVpreds.R
    Modified:   code/getDirectionalDomVarComps.R
    Modified:   code/getGenomicBLUPs.R
    Modified:   code/getMtCrossMeanPreds.R
    Modified:   code/getMtCrossVarPreds.R
    Modified:   code/getUntestedMtCrossVarPreds.R
    Modified:   code/getVarComps.R
    Modified:   data/blups_forawcdata.rds
    Modified:   data/genmap_awc_May2020.rds
    Modified:   data/parentwise_crossVal_folds.rds
    Modified:   data/ped_awc.rds
    Modified:   data/selection_index_weights_4traits.rds
    Modified:   output/CrossesToPredict_top100stdSI_and_209originalParents.rds
    Modified:   output/accuraciesMeans.rds
    Modified:   output/accuraciesUC.rds
    Modified:   output/accuraciesVars.rds
    Modified:   output/crossRealizations/realizedCrossMeans.rds
    Modified:   output/crossRealizations/realizedCrossMeans_BLUPs.rds
    Modified:   output/crossRealizations/realizedCrossMetrics.rds
    Modified:   output/crossRealizations/realizedCrossVars.rds
    Modified:   output/crossRealizations/realizedCrossVars_BLUPs.rds
    Modified:   output/crossRealizations/realized_cross_means_and_covs_traits.rds
    Modified:   output/crossRealizations/realized_cross_means_and_vars_selindices.rds
    Modified:   output/ddEffects.rds
    Modified:   output/gebvs_ModelA_GroupAll_stdSI.rds
    Modified:   output/obsVSpredMeans.rds
    Modified:   output/obsVSpredUC.rds
    Modified:   output/obsVSpredVars.rds
    Modified:   output/pmv_DirectionalDom_varcomps_geneticgroups.rds
    Modified:   output/pmv_varcomps_geneticgroups.rds
    Modified:   output/pmv_varcomps_geneticgroups_tidy_includingSIvars.rds
    Modified:   output/propHomozygous.rds
    Modified:   output/top100stdSI.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Results.Rmd</code>) and HTML (<code>docs/Results.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/2ec2ff8186adf2c50e98b23aee4f119f1a86903f/analysis/Results.Rmd" target="_blank">2ec2ff8</a>
</td>
<td>
wolfemd
</td>
<td>
2021-03-24
</td>
<td>
Update graphics device so each figure saves to unique file name <em>and</em> displays properly in both HTML and PDF outputs.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/PredictOutbredCrossVar/be1e9fcf9d11a80e00a5ee37f8b4970266ded8d0/docs/Results.html" target="_blank">be1e9fc</a>
</td>
<td>
wolfemd
</td>
<td>
2021-03-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/f73b05f90806a89b69b2625c6d64761317eaa00f/analysis/Results.Rmd" target="_blank">f73b05f</a>
</td>
<td>
wolfemd
</td>
<td>
2021-03-24
</td>
<td>
Update to match revised manuscript. Several comparisons moved to Appendix to streamline primary results, figures, etc.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/PredictOutbredCrossVar/4de1330d37b1c2f02a4c9e91970c8675f789d833/docs/Results.html" target="_blank">4de1330</a>
</td>
<td>
wolfemd
</td>
<td>
2021-02-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/883b1d4d2dae0efefd64e4c238477ce34df13895/analysis/Results.Rmd" target="_blank">883b1d4</a>
</td>
<td>
wolfemd
</td>
<td>
2021-02-01
</td>
<td>
Update the syntax highlighting and code-block formatting throughout for
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/6a10c30fbe489268566e4b7568be67a4395c2080/analysis/Results.Rmd" target="_blank">6a10c30</a>
</td>
<td>
wolfemd
</td>
<td>
2021-01-04
</td>
<td>
Submission and GitHub ready version.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/PredictOutbredCrossVar/6a10c30fbe489268566e4b7568be67a4395c2080/docs/Results.html" target="_blank">6a10c30</a>
</td>
<td>
wolfemd
</td>
<td>
2021-01-04
</td>
<td>
Submission and GitHub ready version.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr); <span class="fu">library</span>(ragg); </span></code></pre></div>
<div id="initial-summaries" class="section level1">
<h1>Initial summaries</h1>
<div id="summary-of-the-pedigree-and-germplasm" class="section level2">
<h2>Summary of the pedigree and germplasm</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>ped<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;ped_awc.rds&quot;</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>ped <span class="sc">%&gt;%</span> </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(sireID,damID) <span class="sc">%$%</span> <span class="fu">summary</span>(n)</span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.000   2.000   4.000   6.924  10.000  72.000 </code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>ped <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">c</span>(sireID,damID),<span class="at">names_to =</span> <span class="st">&quot;MaleOrFemale&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Parent&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Parent) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Ncontributions=</span><span class="fu">n</span>()) <span class="sc">%$%</span> <span class="fu">summary</span>(Ncontributions)</span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    4.00   16.00   30.61   36.00  256.00 </code></pre>
<p>There were 3199 comprising 462 families, derived from 209 parents in our pedigree. Parents were used an average of 31 (median 16, range 1-256) times as sire and/or dam in the pedigree. The mean family size was 7 (median 4, range 1-72).</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>propHom<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS14&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(propHom<span class="sc">$</span>PropSNP_homozygous)</span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.7639  0.8254  0.8364  0.8363  0.8476  0.9273 </code></pre>
<p>The average proportion homozygous was 0.84 (range 0.76-0.93) across the 3199 pedigree members (computed over 33370 variable SNP; Table S14).</p>
<p>As expected for a population under recurrent selection, the homozygosity rate increases (though only fractionally) from the C0 (mean 0.826), C1 (0.835), C2 (0.838), C3 (0.839) (Figure S01).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>propHom <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;TMS13|TMS14|TMS15&quot;</span>, GID),<span class="st">&quot;GG (C0)&quot;</span>,<span class="cn">NA</span>),</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TMS13&quot;</span>, GID),<span class="st">&quot;TMS13 (C1)&quot;</span>,Group),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>, GID),<span class="st">&quot;TMS14 (C2)&quot;</span>,Group),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>, GID),<span class="st">&quot;TMS15 (C3)&quot;</span>,Group)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Group) <span class="sc">%&gt;%</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">meanPropHom=</span><span class="fu">round</span>(<span class="fu">mean</span>(PropSNP_homozygous),<span class="dv">3</span>))</span></code></pre></div>
<pre><code># A tibble: 4 x 2
  Group      meanPropHom
  &lt;chr&gt;            &lt;dbl&gt;
1 GG (C0)          0.826
2 TMS13 (C1)       0.835
3 TMS14 (C2)       0.838
4 TMS15 (C3)       0.839</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>propHom <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Group=</span><span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;TMS13|TMS14|TMS15&quot;</span>, GID),<span class="st">&quot;GG (C0)&quot;</span>,<span class="cn">NA</span>),</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TMS13&quot;</span>, GID),<span class="st">&quot;TMS13 (C1)&quot;</span>,Group),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TMS14&quot;</span>, GID),<span class="st">&quot;TMS14 (C2)&quot;</span>,Group),</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Group=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;TMS15&quot;</span>, GID),<span class="st">&quot;TMS15 (C3)&quot;</span>,Group)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.,<span class="fu">aes</span>(<span class="at">x=</span>Group,<span class="at">y=</span>PropSNP_homozygous,<span class="at">fill=</span>Group)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>()</span></code></pre></div>
<p><img src="figure/Results.Rmd/figureS01-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-figureS01-1">
Past versions of figureS01-1.png
</button>
</p>
<div id="fig-figureS01-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/be1e9fcf9d11a80e00a5ee37f8b4970266ded8d0/docs/figure/Results.Rmd/figureS01-1.png" target="_blank">be1e9fc</a>
</td>
<td>
wolfemd
</td>
<td>
2021-03-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/6a10c30fbe489268566e4b7568be67a4395c2080/docs/figure/Results.Rmd/figureS01-1.png" target="_blank">6a10c30</a>
</td>
<td>
wolfemd
</td>
<td>
2021-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="summary-of-the-cross-validation-scheme" class="section level2">
<h2>Summary of the cross-validation scheme</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S2: Summary of cross-validation scheme</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>parentfold_summary<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS02&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>parentfold_summary <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_if</span>(is.numeric,<span class="sc">~</span><span class="fu">ceiling</span>(<span class="fu">mean</span>(.)))</span></code></pre></div>
<pre><code># A tibble: 1 x 4
  Ntestparents Ntrainset Ntestset NcrossesToPredict
         &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;             &lt;dbl&gt;
1           42      1833     1494               167</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>parentfold_summary <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_if</span>(is.numeric,<span class="sc">~</span><span class="fu">ceiling</span>(<span class="fu">min</span>(.))) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Value=</span><span class="st">&quot;Min&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>(parentfold_summary <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">summarize_if</span>(is.numeric,<span class="sc">~</span><span class="fu">ceiling</span>(<span class="fu">max</span>(.))) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Value=</span><span class="st">&quot;Max&quot;</span>))</span></code></pre></div>
<pre><code># A tibble: 2 x 5
  Ntestparents Ntrainset Ntestset NcrossesToPredict Value
         &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;             &lt;dbl&gt; &lt;chr&gt;
1           41      1245     1003               143 Min  
2           42      2323     2081               204 Max  </code></pre>
<p>Across the 5 replications of 5-fold cross-validation the average number of samples was 1833 (range 1245-2323) for training sets and 1494 (range 1003-2081) for testing sets. The 25 training-testing pairs set-up an average of 167 (range 143-204) crosses-to-predict (Table S02).</p>
</div>
<div id="summary-of-the-blups-and-sel.-index" class="section level2">
<h2>Summary of the BLUPs and sel. index</h2>
<p>The correlation between phenotypic BLUPs for the two SI (stdSI and biofortSI; Table S01) was 0.43 (Figure S02). The correlation between DM and TCHART BLUPs, for which we had <em>a priori</em> expectations, was -0.29.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Selection weights -----------</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>indices<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS01&quot;</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BLUPs -----------</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>blups<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;blups_forawcdata.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait,blups) <span class="sc">%&gt;%</span> </span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(blups) <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait,germplasmName,BLUP) <span class="sc">%&gt;%</span> </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Trait,BLUP) <span class="sc">%&gt;%</span> </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(germplasmName,<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;DM&quot;</span>,<span class="st">&quot;logFYLD&quot;</span>,<span class="st">&quot;MCMDS&quot;</span>,<span class="st">&quot;TCHART&quot;</span>)))</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>blups <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(germplasmName,<span class="fu">all_of</span>(indices<span class="sc">$</span>Trait)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">stdSI=</span>blups <span class="sc">%&gt;%</span> </span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(<span class="fu">all_of</span>(indices<span class="sc">$</span>Trait)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span> </span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.matrix</span>(.)<span class="sc">%*%</span>indices<span class="sc">$</span>stdSI,</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">biofortSI=</span>blups <span class="sc">%&gt;%</span> </span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>           <span class="fu">select</span>(<span class="fu">all_of</span>(indices<span class="sc">$</span>Trait)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.data.frame</span>(.) <span class="sc">%&gt;%</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.matrix</span>(.)<span class="sc">%*%</span>indices<span class="sc">$</span>biofortSI)</span></code></pre></div>
<p>Correlations among phenotypic BLUPs (including Selection Indices)</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co">#```{r, fig.show=&quot;hold&quot;, out.width=&quot;50%&quot;}</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>p1<span class="ot">&lt;-</span><span class="fu">ggplot</span>(blups,<span class="fu">aes</span>(<span class="at">x=</span>stdSI,<span class="at">y=</span>biofortSI)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">1.25</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>()</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>corMat<span class="ot">&lt;-</span><span class="fu">cor</span>(blups[,<span class="sc">-</span><span class="dv">1</span>],<span class="at">use =</span> <span class="st">&#39;pairwise.complete.obs&#39;</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>(p1 <span class="sc">|</span> <span class="er">~</span>corrplot<span class="sc">::</span><span class="fu">corrplot</span>(corMat, <span class="at">type =</span> <span class="st">&#39;lower&#39;</span>, <span class="at">col =</span> viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="at">n =</span> <span class="dv">10</span>), <span class="at">diag =</span> T,<span class="at">addCoef.col =</span> <span class="st">&quot;black&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">nrow=</span><span class="dv">1</span>, <span class="at">widths =</span> <span class="fu">c</span>(<span class="fl">0.35</span>,<span class="fl">0.65</span>)) <span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;A&#39;</span>,</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="st">&#39;Correlations among phenotypic BLUPs (including Selection Indices)&#39;</span>)</span></code></pre></div>
<p><img src="figure/Results.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/be1e9fcf9d11a80e00a5ee37f8b4970266ded8d0/docs/figure/Results.Rmd/unnamed-chunk-9-1.png" target="_blank">be1e9fc</a>
</td>
<td>
wolfemd
</td>
<td>
2021-03-24
</td>
</tr>
<tr>
<td>
<a href="https://github.com/wolfemd/PredictOutbredCrossVar/blob/6a10c30fbe489268566e4b7568be67a4395c2080/docs/figure/Results.Rmd/unnamed-chunk-9-1.png" target="_blank">6a10c30</a>
</td>
<td>
wolfemd
</td>
<td>
2021-01-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="predictions-accuracies" class="section level1">
<h1>Predictions accuracies</h1>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Table S10: Accuracies predicting the mean</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>accMeans<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS10&quot;</span>)</span></code></pre></div>
<div id="primary-analysis" class="section level2">
<h2>Primary analysis</h2>
<div id="means" class="section level3">
<h3>Means</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>accMeansMain<span class="ot">&lt;-</span>accMeans <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>, <span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model))</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>accMeansMain <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(predOf,Accuracy) <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>MeanTGV<span class="sc">-</span>MeanBV) <span class="ot">-&gt;</span> diffAcc</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>diffAcc <span class="sc">%$%</span> <span class="fu">summary</span>(diffAcc)</span></code></pre></div>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-0.0826119 -0.0312055 -0.0171228  0.0018702 -0.0004026  0.2795664 </code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>diffAcc <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.05</span>,<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>,<span class="fl">0.95</span>)),</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">diffAcc =</span> <span class="fu">quantile</span>(diffAcc, <span class="fu">c</span>(<span class="fl">0.05</span>,<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>,<span class="fl">0.95</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(quantile,diffAcc)</span></code></pre></div>
<pre><code># A tibble: 1 x 5
    `25%`    `5%`   `50%`     `75%` `95%`
    &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;
1 -0.0312 -0.0656 -0.0171 -0.000403 0.176</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>diffAcc <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.05</span>,<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>,<span class="fl">0.95</span>)),</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">diffAcc =</span> <span class="fu">quantile</span>(diffAcc, <span class="fu">c</span>(<span class="fl">0.05</span>,<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>,<span class="fl">0.95</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(quantile,diffAcc)</span></code></pre></div>
<pre><code># A tibble: 6 x 6
# Groups:   Trait [6]
  Trait       `25%`     `5%`   `50%`    `75%`    `95%`
  &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
1 biofortSI -0.0375 -0.0526  -0.0237 -0.00934  0.0123 
2 DM        -0.0277 -0.0444  -0.0183 -0.0116   0.00924
3 logFYLD    0.0616 -0.00674  0.130   0.180    0.246  
4 MCMDS     -0.0580 -0.0791  -0.0391 -0.0157  -0.00193
5 stdSI     -0.0323 -0.0708  -0.0173  0.00685  0.0271 
6 TCHART    -0.0231 -0.0293  -0.0161 -0.0123  -0.00706</code></pre>
</div>
<div id="covariances" class="section level3">
<h3>(Co)variances</h3>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S11: Accuracies predicting the variances</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>accVars<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS11&quot;</span>)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>accVarsMain<span class="ot">&lt;-</span>accVars <span class="sc">%&gt;%</span> </span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>, <span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>)</span></code></pre></div>
<p>Across traits. Proportion of accuracy estimates above zero? Percentiles: 10%, 50% (median), 90%</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>accVarsMain <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData,<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Component=</span><span class="fu">ifelse</span>(Trait1<span class="sc">==</span>Trait2,<span class="st">&quot;Variance&quot;</span>,<span class="st">&quot;Covariance&quot;</span>),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait1),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Component) <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">propAboveZero=</span><span class="fu">length</span>(<span class="fu">which</span>(AccuracyWtCor<span class="sc">&gt;=</span><span class="dv">0</span>))<span class="sc">/</span><span class="fu">n</span>(),</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)),</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">AccuracyWtCor =</span> <span class="fu">quantile</span>(AccuracyWtCor, <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;quantile&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;AccuracyWtCor&quot;</span>)</span></code></pre></div>
<pre><code># A tibble: 2 x 5
# Groups:   Component [2]
  Component  propAboveZero   `10%`  `50%` `90%`
  &lt;chr&gt;              &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
1 Covariance          0.7  -0.0812 0.0720 0.265
2 Variance            0.89 -0.0118 0.137  0.303</code></pre>
<p>By trait variance / trait-trait covariances.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>accVarsMain <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData,<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Component=</span><span class="fu">ifelse</span>(Trait1<span class="sc">==</span>Trait2,<span class="st">&quot;Variance&quot;</span>,<span class="st">&quot;Covariance&quot;</span>),</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait1),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Component,TraitType,Trait1,Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">propAboveZero=</span><span class="fu">length</span>(<span class="fu">which</span>(AccuracyWtCor<span class="sc">&gt;=</span><span class="dv">0</span>))<span class="sc">/</span><span class="fu">n</span>(),</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)),</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">AccuracyWtCor =</span> <span class="fu">quantile</span>(AccuracyWtCor, <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;quantile&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;AccuracyWtCor&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Component),TraitType,<span class="fu">desc</span>(<span class="st">`</span><span class="at">50%</span><span class="st">`</span>))</span></code></pre></div>
<pre><code># A tibble: 12 x 8
# Groups:   Component, TraitType, Trait1, Trait2 [12]
   Component  TraitType   Trait1  Trait2  propAboveZero    `10%`    `50%`  `90%`
   &lt;chr&gt;      &lt;chr&gt;       &lt;chr&gt;   &lt;chr&gt;           &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;
 1 Variance   ComponentT… MCMDS   MCMDS            0.92  0.0507   0.248   0.360 
 2 Variance   ComponentT… TCHART  TCHART           0.84 -0.0306   0.165   0.283 
 3 Variance   ComponentT… logFYLD logFYLD          1     0.0490   0.150   0.269 
 4 Variance   ComponentT… DM      DM               0.88 -0.0260   0.0651  0.164 
 5 Variance   SI          stdSI   stdSI            0.92  0.0623   0.172   0.283 
 6 Variance   SI          biofor… biofor…          0.78 -0.0501   0.0890  0.194 
 7 Covariance ComponentT… DM      TCHART           0.98  0.115    0.232   0.351 
 8 Covariance ComponentT… logFYLD MCMDS            0.9  -0.00151  0.157   0.297 
 9 Covariance ComponentT… MCMDS   TCHART           0.68 -0.0879   0.0565  0.156 
10 Covariance ComponentT… DM      MCMDS            0.76 -0.0988   0.0515  0.174 
11 Covariance ComponentT… logFYLD TCHART           0.5  -0.142   -0.00147 0.121 
12 Covariance ComponentT… DM      logFYLD          0.38 -0.100   -0.0265  0.0850</code></pre>
<p>Difference in accuracy between VarTGV and VarBV?</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>accVarsMain <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData,<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Component=</span><span class="fu">ifelse</span>(Trait1<span class="sc">==</span>Trait2,<span class="st">&quot;Variance&quot;</span>,<span class="st">&quot;Covariance&quot;</span>),</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait1),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(predOf,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>VarTGV<span class="sc">-</span>VarBV) <span class="ot">-&gt;</span> diffAcc</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>diffAcc <span class="sc">%$%</span> <span class="fu">round</span>(<span class="fu">summary</span>(diffAcc),<span class="dv">3</span>)</span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 -0.174  -0.033  -0.003  -0.002   0.028   0.196 </code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>diffAcc <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)),</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">diffAcc =</span> <span class="fu">quantile</span>(diffAcc, <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(quantile,diffAcc)</span></code></pre></div>
<pre><code># A tibble: 1 x 3
    `10%`    `50%`  `90%`
    &lt;dbl&gt;    &lt;dbl&gt;  &lt;dbl&gt;
1 -0.0634 -0.00303 0.0578</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>diffAcc <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Component,TraitType,Trait1,Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)),</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">diffAcc =</span> <span class="fu">quantile</span>(diffAcc, <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(quantile,diffAcc) <span class="sc">%&gt;%</span> </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Component),TraitType,<span class="fu">desc</span>(<span class="st">`</span><span class="at">50%</span><span class="st">`</span>))</span></code></pre></div>
<pre><code># A tibble: 12 x 7
# Groups:   Component, TraitType, Trait1, Trait2 [12]
   Component  TraitType      Trait1    Trait2      `10%`    `50%`     `90%`
   &lt;chr&gt;      &lt;chr&gt;          &lt;chr&gt;     &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;
 1 Variance   ComponentTrait MCMDS     MCMDS     -0.0111  0.0134   0.0368  
 2 Variance   ComponentTrait TCHART    TCHART    -0.0439 -0.0275  -0.0107  
 3 Variance   ComponentTrait DM        DM        -0.0591 -0.0301  -0.000190
 4 Variance   ComponentTrait logFYLD   logFYLD   -0.0929 -0.0319   0.0908  
 5 Variance   SI             stdSI     stdSI     -0.0284  0.0184   0.0436  
 6 Variance   SI             biofortSI biofortSI -0.0426 -0.00653  0.0262  
 7 Covariance ComponentTrait logFYLD   TCHART    -0.0532  0.0533   0.145   
 8 Covariance ComponentTrait DM        MCMDS     -0.0137  0.0229   0.0563  
 9 Covariance ComponentTrait logFYLD   MCMDS     -0.0372  0.0211   0.0847  
10 Covariance ComponentTrait MCMDS     TCHART    -0.0351  0.0116   0.0384  
11 Covariance ComponentTrait DM        logFYLD   -0.0715  0.00321  0.0626  
12 Covariance ComponentTrait DM        TCHART    -0.0985 -0.0605  -0.0334  </code></pre>
</div>
<div id="usefulness" class="section level3">
<h3>Usefulness</h3>
<p>The usefulness criteria i.e. <span class="math inline">\(UC_{parent}\)</span> and <span class="math inline">\(UC_{variety}\)</span> are predicted by:</p>
<p><span class="math display">\[UC_{parent}=\mu_{BV} + (i_{RS} \times \sigma_{BV})\]</span></p>
<p><span class="math display">\[UC_{variety}=\mu_{TGV} + (i_{VDP} \times \sigma_{TGV})\]</span></p>
<div id="selection-intensity" class="section level4">
<h4>Selection intensity</h4>
<p>In order to combined predicted means and variances into a <strong>UC</strong>, we first calculated the realized intensity of within-family selection (<span class="math inline">\(i_{RS}\)</span> and <span class="math inline">\(i_{VDP}\)</span>). For <span class="math inline">\(UC_{parent}\)</span> we computed the <span class="math inline">\(i_{RS}\)</span> based on the proportion of progeny from each family, that themselves later appeared in the pedigree as parents. For <span class="math inline">\(UC_{clone}\)</span> we compute computed <span class="math inline">\(i_{VDP}\)</span> based on the proportion of family-members that had at least one plot at the penultimate stage of the <strong>VDP</strong>, the advanced yield trial (AYT). For completeness and as part of exploratory analysis, we computed the proportion selected at each of the <strong>VDP</strong> stages: clonal evaluation trial (CET), preliminary yield trial (PYT), advanced yield trial (AYT) and uniform yield trial (UYT).</p>
<p>The table below provides a quick summary of the number of families available with realized selection observed at each stage, plus the corresponding mean selection intensity and proportion selected across families.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S13: Realized within-cross selection metrics</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>crossmetrics<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS13&quot;</span>)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">left_join</span>(crossmetrics <span class="sc">%&gt;%</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>            <span class="fu">select</span>(sireID,damID,<span class="fu">contains</span>(<span class="st">&quot;realIntensity&quot;</span>)) <span class="sc">%&gt;%</span>   distinct <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>            <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">contains</span>(<span class="st">&quot;realIntensity&quot;</span>),</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">names_to =</span> <span class="st">&quot;Stage&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Intensity&quot;</span>,<span class="at">names_prefix =</span> <span class="st">&quot;realIntensity&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>            <span class="fu">summarize</span>(<span class="at">meanIntensity=</span><span class="fu">mean</span>(Intensity, <span class="at">na.rm =</span> T),</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">Nfam=</span><span class="fu">length</span>(<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(Intensity)))),</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>          crossmetrics <span class="sc">%&gt;%</span> </span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>            <span class="fu">select</span>(sireID,damID,<span class="fu">contains</span>(<span class="st">&quot;prop&quot;</span>)) <span class="sc">%&gt;%</span>   distinct <span class="sc">%&gt;%</span> </span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>            <span class="fu">rename</span>(<span class="at">propParent=</span>propUsedAsParent,</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">propCET=</span>propPhenotyped,</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">propPYT=</span>propPastCET,</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">propAYT=</span>propPastPYT,</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">propUYT=</span>propPastAYT) <span class="sc">%&gt;%</span> </span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>            <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">contains</span>(<span class="st">&quot;prop&quot;</span>),</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">values_to =</span> <span class="st">&quot;PropPast&quot;</span>,<span class="at">names_to =</span> <span class="st">&quot;Stage&quot;</span>,<span class="at">names_prefix =</span> <span class="st">&quot;propPast|prop&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>            <span class="fu">group_by</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>            <span class="fu">summarize</span>(<span class="at">meanPropPast=</span><span class="fu">mean</span>(PropPast, <span class="at">na.rm =</span> T))) <span class="sc">%&gt;%</span></span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Stage=</span><span class="fu">factor</span>(Stage,<span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Parent&quot;</span>,<span class="st">&quot;CET&quot;</span>,<span class="st">&quot;PYT&quot;</span>,<span class="st">&quot;AYT&quot;</span>,<span class="st">&quot;UYT&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Stage,Nfam,meanIntensity,meanPropPast) <span class="sc">%&gt;%</span> <span class="fu">mutate_if</span>(is.numeric,<span class="sc">~</span><span class="fu">round</span>(.,<span class="dv">2</span>))</span></code></pre></div>
<pre><code># A tibble: 5 x 4
  Stage   Nfam meanIntensity meanPropPast
  &lt;fct&gt;  &lt;dbl&gt;         &lt;dbl&gt;        &lt;dbl&gt;
1 Parent    48          1.59         0.02
2 CET      415          0.26         0.76
3 PYT      288          0.9          0.3 
4 AYT      104          1.46         0.05
5 UYT       39          1.4          0.03</code></pre>
<p>There were 48 families with a mean intensity of 1.59 (mean 2% selected) that themselves had members who were parents in the pedigree. As expected, the number of available families and the proportion selected decreased (increasing selection intensity) from CET to UYT. We choose to focus on the AYT stage, which has 104 families, mean intensity 1.46 (mean 5% selected).</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S9: Predicted and observed UC</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>predVSobsUC<span class="ot">&lt;-</span><span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTable09.csv&quot;</span>),<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>uc_cv_summary<span class="ot">&lt;-</span>predVSobsUC <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Model,predOf,Stage,Trait,Repeat,Fold) <span class="sc">%&gt;%</span> </span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Nfam=</span><span class="fu">n</span>(),</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanFamSize=</span><span class="fu">round</span>(<span class="fu">mean</span>(FamSize),<span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Trait,<span class="sc">-</span>Model) <span class="sc">%&gt;%</span> </span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>  distinct</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>uc_cv_summary <span class="sc">%&gt;%</span> </span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(predOf,Stage) <span class="sc">%&gt;%</span> </span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">minNfam=</span><span class="fu">min</span>(Nfam),</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanNfam=</span><span class="fu">mean</span>(Nfam),</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">maxNfam=</span><span class="fu">max</span>(Nfam),</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>            <span class="at">minMeanFamSize=</span><span class="fu">min</span>(meanFamSize),</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanMeanFamSize=</span><span class="fu">mean</span>(meanFamSize),</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>            <span class="at">maxMeanFamSize=</span><span class="fu">max</span>(meanFamSize))</span></code></pre></div>
<pre><code># A tibble: 7 x 8
# Groups:   predOf [2]
  predOf Stage          minNfam meanNfam maxNfam minMeanFamSize meanMeanFamSize
  &lt;chr&gt;  &lt;chr&gt;            &lt;int&gt;    &lt;dbl&gt;   &lt;int&gt;          &lt;dbl&gt;           &lt;dbl&gt;
1 BV     ConstIntensity     143    166.      204            5.9            6.85
2 BV     Parent               9     17.3      24            8.8           13.8 
3 TGV    AYT                 25     37.3      50           10.9           12.9 
4 TGV    CET                129    149.      192            6.3            7.48
5 TGV    ConstIntensity     143    166.      204            5.9            6.85
6 TGV    PYT                 69    103.      142            7.3            8.73
7 TGV    UYT                  8     14.1      20            7.3           11.8 
# … with 1 more variable: maxMeanFamSize &lt;dbl&gt;</code></pre>
</div>
<div id="accuracy-comparisons" class="section level4">
<h4>Accuracy comparisons</h4>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S12: Accuracies predicting the usefulness criteria</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>accUC<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS12&quot;</span>)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>accUCmain<span class="ot">&lt;-</span>accUC <span class="sc">%&gt;%</span> </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>,<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),Stage <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Parent&quot;</span>,<span class="st">&quot;AYT&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>VarMethod,<span class="sc">-</span>AccuracyCor,<span class="sc">-</span>Stage)</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co"># accMeansMain %&gt;% count(Trait,predOf)</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co"># accVarsMain %&gt;% count(Trait1,Trait2,predOf)</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="co"># accUCmain %&gt;% count(Repeat,Fold)</span></span></code></pre></div>
<p>Proportion of UC accuracy estimates greater than zero.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(accUC<span class="sc">$</span>AccuracyWtCor<span class="sc">&gt;</span><span class="dv">0</span>))<span class="sc">/</span><span class="fu">nrow</span>(accUC)</span></code></pre></div>
<pre><code>[1] 0.952619</code></pre>
<p>Compare Mean, Variance and UC accuracy:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>meanVsUCacc<span class="ot">&lt;-</span>accUCmain <span class="sc">%&gt;%</span> </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">AccUC=</span>AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(accMeansMain <span class="sc">%&gt;%</span> </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>ValidationData,<span class="sc">-</span>Model) <span class="sc">%&gt;%</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">predOf=</span><span class="fu">gsub</span>(<span class="st">&quot;Mean&quot;</span>,<span class="st">&quot;&quot;</span>,predOf)) <span class="sc">%&gt;%</span> </span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">AccMean=</span>Accuracy))</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>meanVsUCacc <span class="sc">%&lt;&gt;%</span> </span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(accVarsMain <span class="sc">%&gt;%</span> </span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(Trait1<span class="sc">==</span>Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>ValidationData,<span class="sc">-</span>Model,<span class="sc">-</span>AccuracyCor,<span class="sc">-</span>VarMethod,<span class="sc">-</span>Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>              <span class="fu">mutate</span>(<span class="at">predOf=</span><span class="fu">gsub</span>(<span class="st">&quot;Var&quot;</span>,<span class="st">&quot;&quot;</span>,predOf)) <span class="sc">%&gt;%</span> </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">AccVar=</span>AccuracyWtCor,</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>                     <span class="at">Trait=</span>Trait1))</span></code></pre></div>
<p>Correlation between mean, Var and UC accuracy, overall</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>meanVsUCacc <span class="sc">%&gt;%</span> </span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corMeanUCacc=</span><span class="fu">cor</span>(AccUC,AccMean),</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanDiffMeanUC=</span><span class="fu">mean</span>(AccUC<span class="sc">-</span>AccMean),</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">corMeanVarAcc=</span><span class="fu">cor</span>(AccVar,AccMean),</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">corVarUCacc=</span><span class="fu">cor</span>(AccUC,AccVar))</span></code></pre></div>
<pre><code># A tibble: 1 x 4
  corMeanUCacc meanDiffMeanUC corMeanVarAcc corVarUCacc
         &lt;dbl&gt;          &lt;dbl&gt;         &lt;dbl&gt;       &lt;dbl&gt;
1        0.745        -0.0929        -0.222      -0.236</code></pre>
<p><strong>Correlation between mean, Var and UC accuracy, by predOf (BV vs. TGV)</strong></p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>meanVsUCacc <span class="sc">%&gt;%</span> </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(predOf) <span class="sc">%&gt;%</span> </span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corMeanUCacc=</span><span class="fu">cor</span>(AccUC,AccMean),</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanDiffMeanUC=</span><span class="fu">mean</span>(AccUC<span class="sc">-</span>AccMean),</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">corMeanVarAcc=</span><span class="fu">cor</span>(AccVar,AccMean))</span></code></pre></div>
<pre><code># A tibble: 2 x 4
  predOf corMeanUCacc meanDiffMeanUC corMeanVarAcc
  &lt;chr&gt;         &lt;dbl&gt;          &lt;dbl&gt;         &lt;dbl&gt;
1 BV            0.704        -0.0748        -0.177
2 TGV           0.802        -0.111         -0.275</code></pre>
<p><strong>Correlation between mean, Var and UC accuracy, by Trait</strong></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>meanVsUCacc <span class="sc">%&gt;%</span> </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corMeanUCacc=</span><span class="fu">cor</span>(AccUC,AccMean),</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanDiffMeanUC=</span><span class="fu">mean</span>(AccUC<span class="sc">-</span>AccMean),</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">corMeanVarAcc=</span><span class="fu">cor</span>(AccVar,AccMean)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(corMeanUCacc))</span></code></pre></div>
<pre><code># A tibble: 6 x 4
  Trait     corMeanUCacc meanDiffMeanUC corMeanVarAcc
  &lt;chr&gt;            &lt;dbl&gt;          &lt;dbl&gt;         &lt;dbl&gt;
1 TCHART          0.494        -0.00356      -0.0951 
2 logFYLD         0.410        -0.0762       -0.0571 
3 MCMDS           0.358        -0.280         0.475  
4 stdSI           0.315        -0.0736        0.129  
5 DM              0.110        -0.0938        0.227  
6 biofortSI       0.0189       -0.0297        0.00484</code></pre>
<p><strong>Make an overall comparison of trait accuracies for UC</strong></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>accUCmain <span class="sc">%&gt;%</span> </span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">propAboveZero=</span><span class="fu">length</span>(<span class="fu">which</span>(AccuracyWtCor<span class="sc">&gt;</span><span class="dv">0</span>))<span class="sc">/</span><span class="fu">n</span>(),</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)),</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">AccuracyWtCor =</span> <span class="fu">quantile</span>(AccuracyWtCor, <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;quantile&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;AccuracyWtCor&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(is.numeric,<span class="sc">~</span><span class="fu">round</span>(.,<span class="dv">2</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="st">`</span><span class="at">50%</span><span class="st">`</span>))</span></code></pre></div>
<pre><code># A tibble: 6 x 5
# Groups:   Trait [6]
  Trait     propAboveZero `10%` `50%` `90%`
  &lt;chr&gt;             &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
1 TCHART             1     0.66 0.83   0.94
2 DM                 1     0.36 0.65   0.81
3 biofortSI          1     0.39 0.580  0.74
4 stdSI              0.96  0.16 0.49   0.65
5 logFYLD            0.88 -0.04 0.24   0.49
6 MCMDS              0.7  -0.18 0.1    0.41</code></pre>
<p>Compare UC accuracy of TGV vs. BV, overall</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>accUCmain <span class="sc">%&gt;%</span> </span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(predOf,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>TGV<span class="sc">-</span>BV) <span class="sc">%$%</span> <span class="fu">round</span>(<span class="fu">summary</span>(diffAcc),<span class="dv">2</span>)</span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  -0.75   -0.17   -0.06   -0.03    0.06    1.01 </code></pre>
<p>Proportion of accuracy estimates where BV&gt;TGV.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>accUCmain <span class="sc">%&gt;%</span> </span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(predOf,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>TGV<span class="sc">-</span>BV) <span class="sc">%$%</span> (<span class="fu">length</span>(<span class="fu">which</span>(diffAcc<span class="sc">&lt;=</span><span class="dv">0</span>))<span class="sc">/</span><span class="fu">length</span>(diffAcc))</span></code></pre></div>
<pre><code>[1] 0.62</code></pre>
<p>Compare UC accuracy difference TGV minus BV, by Trait</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>accUCmain <span class="sc">%&gt;%</span> </span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(predOf,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>TGV<span class="sc">-</span>BV) <span class="sc">%&gt;%</span> </span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait,TraitType) <span class="sc">%&gt;%</span> </span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">quantile =</span> scales<span class="sc">::</span><span class="fu">percent</span>(<span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>)),</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">diffAcc =</span> <span class="fu">quantile</span>(diffAcc, <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.5</span>,<span class="fl">0.9</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;quantile&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;diffAcc&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(is.numeric,<span class="sc">~</span><span class="fu">round</span>(.,<span class="dv">2</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="st">`</span><span class="at">50%</span><span class="st">`</span>))</span></code></pre></div>
<pre><code># A tibble: 6 x 5
# Groups:   Trait, TraitType [6]
  Trait     TraitType      `10%` `50%` `90%`
  &lt;chr&gt;     &lt;chr&gt;          &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
1 stdSI     SI             -0.15  0.1   0.44
2 logFYLD   ComponentTrait -0.26 -0.02  0.28
3 DM        ComponentTrait -0.24 -0.05  0.12
4 biofortSI SI             -0.19 -0.08  0.12
5 TCHART    ComponentTrait -0.17 -0.08  0.07
6 MCMDS     ComponentTrait -0.49 -0.17  0.12</code></pre>
<ul>
<li>Mean: TGV&lt;BV (median -0.017)</li>
<li>Var: No consistent / small differences</li>
<li>UC: TGV&lt;BV</li>
</ul>
</div>
</div>
</div>
</div>
<div id="population-estimates-of-the-importance-of-dominance-variance" class="section level1">
<h1>Population estimates of the importance of dominance variance</h1>
<p>In this study, our focus is mainly on distinguishing among crosses, and the accuracy of cross-metric predictions. Detailed analysis of the additive-dominance genetic variance-covariance structure in cassava (sub)-populations is an important topic, which we mostly leave for future study. However, we make a brief examination of the genetic variance-covariance estimates associated with the overall population and component genetic groups. We report all variance-covariance estimates in <strong>TableS15</strong> and complete BGLR output in the <a href="ftp://ftp.cassavabase.org/marnin_datasets/Wolfe_et_al_GenomicMateSelection">FTP repository associated with this study</a>.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S15: Variance estimates for genetic groups</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>varcomps<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS15&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>, Method<span class="sc">==</span><span class="st">&quot;M2&quot;</span>,Model <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;DirDomAD&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>VarMethod,<span class="sc">-</span>Method)</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>varcomps <span class="sc">%&gt;%</span> </span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Trait1<span class="sc">==</span>Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Model) <span class="sc">%&gt;%</span> </span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">minPropDom=</span><span class="fu">min</span>(propDom),</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanPropDom=</span><span class="fu">mean</span>(propDom),</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">maxPropDom=</span><span class="fu">max</span>(propDom))</span></code></pre></div>
<pre><code># A tibble: 1 x 4
  Model    minPropDom meanPropDom maxPropDom
  &lt;chr&gt;         &lt;dbl&gt;       &lt;dbl&gt;      &lt;dbl&gt;
1 DirDomAD       0.06       0.236       0.53</code></pre>
<p>Over all genetic groups analyzed, across trait and SI variances, dominance accounted for an average of 24% (range 6-53%).</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>varcomps <span class="sc">%&gt;%</span> </span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Trait1<span class="sc">==</span>Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait1,Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">minPropDom=</span><span class="fu">min</span>(propDom),</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">meanPropDom=</span><span class="fu">mean</span>(propDom),</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">maxPropDom=</span><span class="fu">max</span>(propDom)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(meanPropDom))</span></code></pre></div>
<pre><code># A tibble: 6 x 5
# Groups:   Trait1 [6]
  Trait1    Trait2    minPropDom meanPropDom maxPropDom
  &lt;chr&gt;     &lt;chr&gt;          &lt;dbl&gt;       &lt;dbl&gt;      &lt;dbl&gt;
1 logFYLD   logFYLD         0.41       0.462       0.53
2 biofortSI biofortSI       0.18       0.264       0.5 
3 stdSI     stdSI           0.14       0.232       0.35
4 DM        DM              0.11       0.18        0.27
5 MCMDS     MCMDS           0.08       0.172       0.3 
6 TCHART    TCHART          0.06       0.106       0.21</code></pre>
<p>Dominance was most important (mean 46% of genetic variance) for yield (logFYLD) and least important for TCHART (mean 11%) (Figure 4).</p>
</div>
<div id="population-estimates-of-inbreeding-effects" class="section level1">
<h1>Population estimates of inbreeding effects</h1>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S16: Directional dominance effects estimates</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>ddEffects<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS16&quot;</span>)</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>ddEffects <span class="sc">%&gt;%</span> </span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Repeat,<span class="sc">-</span>Fold,<span class="sc">-</span>InbreedingEffectSD) <span class="sc">%&gt;%</span> </span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Dataset=</span><span class="fu">ifelse</span>(Dataset<span class="sc">!=</span><span class="st">&quot;GeneticGroups&quot;</span>,Group,Dataset)) <span class="sc">%&gt;%</span> </span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Group) <span class="sc">%&gt;%</span> </span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Dataset,Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_all</span>(<span class="sc">~</span><span class="fu">round</span>(<span class="fu">mean</span>(.),<span class="dv">3</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Dataset,InbreedingEffect)</span></code></pre></div>
<pre><code># A tibble: 4 x 3
  Trait   GeneticGroups ParentwiseCV
  &lt;chr&gt;           &lt;dbl&gt;        &lt;dbl&gt;
1 DM             -4.82        -7.85 
2 logFYLD        -2.75        -3.88 
3 MCMDS           0.32         1.27 
4 TCHART         -0.004       -0.015</code></pre>
<p>We found mostly consistent and significant (diff. from zero) effects of inbreeding <em>depression</em> associated especially (<strong>Figure 5</strong>, <strong>Table S16</strong>), with logFYLD (mean effect -2.75 across genetic groups, -3.88 across cross-validation folds), but also DM (-4.82 genetic groups, -7.85 cross-validation) and MCMDS (0.32 genetic groups, 1.27 cross-validation). This corresponds to higher homozygosity being associated with lower DM, lower yield and worse disease.</p>
</div>
<div id="exploring-untested-crosses" class="section level1">
<h1>Exploring Untested Crosses</h1>
<p>We made 8 predictions (2 SIs x 2 prediction targets [BV, TGV] x 2 criteria [Mean, UC = Mean + 2*SD]) prediction for each of 47,083 possible crosses of 306 parents.</p>
<div id="correlations-among-predictions" class="section level2">
<h2>Correlations among predictions</h2>
<p>First, quickly evaluate the multivariate decision space encompassed by predictions of mean, SD, UC for BV and TGV.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr); </span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses<span class="ot">&lt;-</span><span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTable18.csv&quot;</span>),<span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Model<span class="sc">==</span><span class="st">&quot;DirDom&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>Model)</span></code></pre></div>
<p>Average correlations between BiofortSI and StdSI by prediction.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Trait,Pred) <span class="sc">%&gt;%</span> </span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(PredOf,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corSelIndices=</span><span class="fu">cor</span>(stdSI,biofortSI)) <span class="sc">%&gt;%</span> </span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(PredOf) <span class="sc">%&gt;%</span> </span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">meanCorSIs=</span><span class="fu">mean</span>(corSelIndices))</span></code></pre></div>
<pre><code># A tibble: 3 x 2
  PredOf meanCorSIs
  &lt;chr&gt;       &lt;dbl&gt;
1 Mean        0.204
2 Sd          0.912
3 UC          0.136</code></pre>
<p><strong>TABLE:</strong> Correlations between predictions about each selection index (<span class="math inline">\(\overset{StdSI,BiofortSI}{\textbf{cor}}\)</span>).</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Trait,Pred) <span class="sc">%&gt;%</span> </span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(PredOf,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corSelIndices=</span><span class="fu">cor</span>(stdSI,biofortSI)) <span class="sc">%&gt;%</span> </span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Component,corSelIndices) <span class="sc">%&gt;%</span> </span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(PredOf) <span class="sc">%&gt;%</span> </span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["PredOf"],"name":[1],"type":["chr"],"align":["left"]},{"label":["BV"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["TGV"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"Mean","2":"0.2202287","3":"0.1882594"},{"1":"Sd","2":"0.8825580","3":"0.9420331"},{"1":"UC","2":"0.1640598","3":"0.1080554"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p><strong>TABLE:</strong> Correlations between predictions about each component, for each trait (<span class="math inline">\(\overset{BV,TGV}{\textbf{cor}}\)</span>).</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Component,Pred) <span class="sc">%&gt;%</span> </span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait,PredOf) <span class="sc">%&gt;%</span> </span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corBV_TGV=</span><span class="fu">round</span>(<span class="fu">cor</span>(BV,TGV),<span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Trait,corBV_TGV)</span></code></pre></div>
<pre><code># A tibble: 3 x 3
  PredOf biofortSI stdSI
  &lt;chr&gt;      &lt;dbl&gt; &lt;dbl&gt;
1 Mean        0.96  0.95
2 Sd          0.91  0.88
3 UC          0.95  0.93</code></pre>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(PredOf,Pred) <span class="sc">%&gt;%</span> </span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corMeanSD=</span><span class="fu">round</span>(<span class="fu">cor</span>(Mean,Sd),<span class="dv">2</span>),</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">corMeanUC=</span><span class="fu">round</span>(<span class="fu">cor</span>(Mean,UC),<span class="dv">2</span>),</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">corSdUC=</span><span class="fu">round</span>(<span class="fu">cor</span>(Sd,UC),<span class="dv">2</span>)) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summarize</span>(<span class="fu">across</span>(is.numeric,mean))<span class="co">#corComponents=round(cor(BV,TGV),2)) %$% summary(corComponents) </span></span></code></pre></div>
<pre><code># A tibble: 1 x 3
  corMeanSD corMeanUC corSdUC
      &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;
1    -0.365     0.995  -0.258</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(PredOf,Pred) <span class="sc">%&gt;%</span> </span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corMeanSD=</span><span class="fu">round</span>(<span class="fu">cor</span>(Mean,Sd),<span class="dv">2</span>),</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">corMeanUC=</span><span class="fu">round</span>(<span class="fu">cor</span>(Mean,UC),<span class="dv">2</span>),</span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">corSdUC=</span><span class="fu">round</span>(<span class="fu">cor</span>(Sd,UC),<span class="dv">2</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>  rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Trait"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Component"],"name":[2],"type":["chr"],"align":["left"]},{"label":["corMeanSD"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["corMeanUC"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["corSdUC"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"biofortSI","2":"BV","3":"-0.23","4":"1.00","5":"-0.13"},{"1":"biofortSI","2":"TGV","3":"-0.28","4":"0.99","5":"-0.16"},{"1":"stdSI","2":"BV","3":"-0.48","4":"1.00","5":"-0.40"},{"1":"stdSI","2":"TGV","3":"-0.47","4":"0.99","5":"-0.34"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>The mean and variance have a low, but negative correlation. At the standardized intensity of 2.67 (1% selected), leads to a small negative correlation between SD and UC. The crosses with highest mean will mostly be those with highest UC. The crosses with highest mean will also have a small tendency to have smaller variance.</p>
<p>Nevertheless, the <strong>biggest differences</strong> in decision space have to do with the difference between using the Mean vs. including the SD via the UC.</p>
</div>
<div id="decision-space---top-50-crosses" class="section level2">
<h2>Decision space - top 50 crosses?</h2>
<p>What we next want to know, is how different the selections of crosses-to-make would be if we use different criteria, particularly the mean vs. the UC.</p>
<p>For each of the 8 predictions of 47,083 crosses, select the top 50 ranked crosses.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>top50crosses<span class="ot">&lt;-</span>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PredOf<span class="sc">!=</span><span class="st">&quot;Sd&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait,PredOf,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">order_by =</span> Pred,<span class="at">n=</span><span class="dv">50</span>) <span class="sc">%&gt;%</span> <span class="fu">ungroup</span>()</span></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(sireID,damID) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span></code></pre></div>
<pre><code>[1] 202</code></pre>
<p>Number of distinct crosses selected per Trait</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait,sireID,damID) <span class="sc">%&gt;%</span> </span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Trait)</span></code></pre></div>
<pre><code># A tibble: 2 x 2
  Trait         n
  &lt;chr&gt;     &lt;int&gt;
1 biofortSI    90
2 stdSI       112</code></pre>
<p>Number of Self vs. Outcross selected by Trait</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait,sireID,damID,IsSelf) <span class="sc">%&gt;%</span> </span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Trait,IsSelf)</span></code></pre></div>
<pre><code># A tibble: 4 x 3
  Trait     IsSelf     n
  &lt;chr&gt;     &lt;lgl&gt;  &lt;int&gt;
1 biofortSI FALSE     84
2 biofortSI TRUE       6
3 stdSI     FALSE    105
4 stdSI     TRUE       7</code></pre>
<p>Only 202 unique crosses selected based on at least one of the 8 criteria. Of those 112 were selected for the StdSI (90 Biofort) including 7 (6) selfs on the StdSI (BiofortSI).</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait,sireID,damID) <span class="sc">%&gt;%</span> </span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Selected=</span><span class="st">&quot;Yes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Trait,Selected) <span class="sc">%&gt;%</span> </span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>(.)</span></code></pre></div>
<pre><code># A tibble: 0 x 4
# … with 4 variables: sireID &lt;chr&gt;, damID &lt;chr&gt;, biofortSI &lt;chr&gt;, stdSI &lt;chr&gt;</code></pre>
<p>There were 0 crosses selected for both SI.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(CrossPrevMade,sireID,damID) <span class="sc">%&gt;%</span> </span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(CrossPrevMade)</span></code></pre></div>
<pre><code># A tibble: 1 x 2
  CrossPrevMade     n
  &lt;chr&gt;         &lt;int&gt;
1 No              202</code></pre>
<p>None of the selected crosses have previously been tested.</p>
<p><strong>Table:</strong> Summarize, by trait, the number of and relative contributions (number of matings) proposed for each parent selected in the group of top crosses.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Family=</span><span class="fu">paste0</span>(sireID,<span class="st">&quot;x&quot;</span>,damID)) <span class="sc">%&gt;%</span> </span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Trait,Family,sireID,damID) <span class="sc">%&gt;%</span> </span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(sireID,damID), <span class="at">names_to =</span> <span class="st">&quot;Parent&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;germplasmName&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Trait,germplasmName) <span class="sc">%&gt;%</span> </span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Nparents=</span><span class="fu">length</span>(<span class="fu">unique</span>(germplasmName)),</span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">minProg=</span><span class="fu">min</span>(n),<span class="at">maxProg=</span><span class="fu">max</span>(n),<span class="at">medianProg=</span><span class="fu">median</span>(n))</span></code></pre></div>
<pre><code># A tibble: 2 x 5
  Trait     Nparents minProg maxProg medianProg
  &lt;chr&gt;        &lt;int&gt;   &lt;int&gt;   &lt;int&gt;      &lt;dbl&gt;
1 biofortSI       33       1      81          4
2 stdSI           44       1      70          3</code></pre>
<p><strong>Next:</strong> For each SI, break down the criteria for which the “best” crosses are interesting.</p>
<p>Quantify the number of unique crosses selected by:</p>
<p><strong>Component (BV vs. TGV)</strong></p>
<ul>
<li>Not many crosses are selected for both their BV <em>and</em> TGV?</li>
<li>Selfs get selected mostly by TGV???</li>
</ul>
<p><strong>Table:</strong> Number of crosses for each SI selected by BV, TGV or both, ignoring selection based Mean vs. UC.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait,sireID,damID,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Selected=</span><span class="st">&quot;Yes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Component,Selected) <span class="sc">%&gt;%</span> </span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(.),replace_na,<span class="at">replace =</span> <span class="st">&quot;No&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Trait,BV,TGV) <span class="sc">%&gt;%</span> <span class="fu">spread</span>(Trait,n) <span class="sc">%&gt;%</span> </span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["BV"],"name":[1],"type":["chr"],"align":["left"]},{"label":["TGV"],"name":[2],"type":["chr"],"align":["left"]},{"label":["biofortSI"],"name":[3],"type":["int"],"align":["right"]},{"label":["stdSI"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"No","2":"Yes","3":"29","4":"48"},{"1":"Yes","2":"No","3":"31","4":"47"},{"1":"Yes","2":"Yes","3":"30","4":"17"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p><strong>Table:</strong> Number of crosses for each SI selected by BV, TGV or both, ignoring selection based Mean vs. UC. Broken down by whether</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> <span class="fu">filter</span>(IsSelf<span class="sc">==</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait,sireID,damID,Component) <span class="sc">%&gt;%</span> </span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Selected=</span><span class="st">&quot;Yes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Component,Selected) <span class="sc">%&gt;%</span> </span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(.),replace_na,<span class="at">replace =</span> <span class="st">&quot;No&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Trait,BV) <span class="sc">%&gt;%</span> </span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>  rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Trait"],"name":[1],"type":["chr"],"align":["left"]},{"label":["BV"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"biofortSI","2":"Yes","3":"6"},{"1":"stdSI","2":"Yes","3":"7"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Selfs were <em>only</em> selected based on BV.</p>
<p><strong>Table:</strong> Compute the number of parents unique selected based on BV vs. TGV</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nest</span>(<span class="at">families=</span><span class="fu">c</span>(<span class="sc">-</span>Trait,<span class="sc">-</span>Component)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Component,families) <span class="sc">%&gt;%</span> </span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">NparentsBVunique=</span><span class="fu">map2_dbl</span>(BV,TGV,<span class="sc">~</span><span class="fu">length</span>(<span class="fu">union</span>(.x<span class="sc">$</span>sireID,.x<span class="sc">$</span>damID) <span class="sc">%&gt;%</span> .[<span class="sc">!</span>. <span class="sc">%in%</span> <span class="fu">union</span>(.y<span class="sc">$</span>sireID,.y<span class="sc">$</span>damID)])),</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">NparentsTGVunique=</span><span class="fu">map2_dbl</span>(BV,TGV,<span class="sc">~</span><span class="fu">length</span>(<span class="fu">union</span>(.y<span class="sc">$</span>sireID,.y<span class="sc">$</span>damID) <span class="sc">%&gt;%</span> .[<span class="sc">!</span>. <span class="sc">%in%</span> <span class="fu">union</span>(.x<span class="sc">$</span>sireID,.x<span class="sc">$</span>damID)])),</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">NparentsTot=</span><span class="fu">map2_dbl</span>(BV,TGV,<span class="sc">~</span><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">c</span>(.x<span class="sc">$</span>sireID,.x<span class="sc">$</span>damID,.y<span class="sc">$</span>sireID,.y<span class="sc">$</span>damID))))) <span class="sc">%&gt;%</span> </span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>BV,<span class="sc">-</span>TGV) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(Trait) <span class="sc">%&gt;%</span> rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Trait"],"name":[1],"type":["chr"],"align":["left"]},{"label":["NparentsBVunique"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["NparentsTGVunique"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["NparentsTot"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"biofortSI","2":"1","3":"14","4":"33"},{"1":"stdSI","2":"4","3":"22","4":"44"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p><strong>Table:</strong> Compute the number of parents uniquely selected based on Mean, UC and both</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(Trait,sireID,damID,PredOf) <span class="sc">%&gt;%</span> </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Selected=</span><span class="st">&quot;Yes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(PredOf,Selected) <span class="sc">%&gt;%</span> </span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(.),replace_na,<span class="at">replace =</span> <span class="st">&quot;No&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Trait,Mean,UC) <span class="sc">%&gt;%</span><span class="fu">spread</span>(Trait,n) <span class="sc">%&gt;%</span> </span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  rmarkdown<span class="sc">::</span><span class="fu">paged_table</span>()</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Mean"],"name":[1],"type":["chr"],"align":["left"]},{"label":["UC"],"name":[2],"type":["chr"],"align":["left"]},{"label":["biofortSI"],"name":[3],"type":["int"],"align":["right"]},{"label":["stdSI"],"name":[4],"type":["int"],"align":["right"]}],"data":[{"1":"No","2":"Yes","3":"14","4":"27"},{"1":"Yes","2":"No","3":"14","4":"28"},{"1":"Yes","2":"Yes","3":"62","4":"57"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="so-which-are-the-best-crosses" class="section level2">
<h2>So which are the “BEST” crosses?</h2>
<ul>
<li>Chosen most times, for most criteria?</li>
</ul>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>best50crosses<span class="ot">&lt;-</span>top50crosses <span class="sc">%&gt;%</span> </span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(sireID,damID,Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span> </span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_max</span>(<span class="at">order_by =</span> <span class="st">`</span><span class="at">n</span><span class="st">`</span>,<span class="at">n =</span> <span class="dv">50</span>, <span class="at">with_ties =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">NtimesChosen=</span>n) </span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>best50crosses <span class="sc">%&gt;%</span> <span class="fu">count</span>(Trait)</span></code></pre></div>
<pre><code># A tibble: 2 x 2
# Groups:   Trait [2]
  Trait         n
  &lt;chr&gt;     &lt;int&gt;
1 biofortSI    66
2 stdSI        58</code></pre>
<p>If you use the most times chosen as the criteria and you don’t break ties, there are 58 StdSI and 66 BiofortSI crosses to consider as the “best”.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Are these crosses really &quot;better&quot;?</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="co"># bind_rows(predUntestedCrosses %&gt;% </span></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="co">#             filter(CrossPrevMade==&quot;Yes&quot;) %&gt;% </span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="co">#             mutate(Group=ifelse(grepl(&quot;TMS13&quot;,sireID) &amp; grepl(&quot;TMS13&quot;,damID),&quot;C1&quot;,</span></span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                                 ifelse(grepl(&quot;TMS14&quot;,sireID) &amp; grepl(&quot;TMS14&quot;,damID),&quot;C2&quot;,</span></span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                                        ifelse(grepl(&quot;TMS15&quot;,sireID) &amp; grepl(&quot;TMS15&quot;,damID),&quot;C3&quot;,&quot;C0&quot;)))),</span></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="co">#           predUntestedCrosses %&gt;% </span></span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a><span class="co">#             semi_join(best50crosses) %&gt;% </span></span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a><span class="co">#             mutate(Group=&quot;NewCrosses&quot;)) %&gt;% </span></span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(Group,Trait,PredOf,Component) %&gt;% </span></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarize(meanPred=mean(Pred),</span></span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a><span class="co">#             sePred=sd(Pred)/sqrt(n())) %&gt;% </span></span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(.,aes(x=Group,y=meanPred,fill=Component)) + </span></span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_grid(PredOf~Trait, scales = &#39;free&#39;) + </span></span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_bar(stat = &#39;identity&#39;, position=&#39;dodge&#39;)</span></span></code></pre></div>
</div>
</div>
<div id="supplementary-questions" class="section level1">
<h1>Supplementary Questions</h1>
<div id="does-the-validation-type-make-a-difference" class="section level2">
<h2>Does the validation type make a difference?</h2>
<p>Most often, cross-validation done to test genomic prediction accuracy uses validation data (the stand-in for “truth”) consisting of adjusted values, (e.g. BLUPs or BLUEs) for total individual performance, not including genomic relatedness information. In our study we set-up cross-validation folds that enable use to predict the GEBV and GETGV (GBLUPs) of validation family-members, and to subsequently compute their sample means, variances and usefulness. This approach has the added advantage of expanding the available sample size of validation progeny with complete data across traits. Nevertheless, we made some comparison to results using BLUPs that do not incorporate genomic relatedness information; in other words, independent and identically distributed (i.i.d.) BLUPs.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Table S10: Accuracies predicting the mean</span></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>accMeans<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS10&quot;</span>)</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S11: Accuracies predicting the variances</span></span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>accVars<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS11&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Component=</span><span class="fu">ifelse</span>(Trait1<span class="sc">==</span>Trait2,<span class="st">&quot;Variance&quot;</span>,<span class="st">&quot;Covariance&quot;</span>),</span>
<span id="cb94-7"><a href="#cb94-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait1),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>))</span></code></pre></div>
<p><strong>Summarize accuracy differences between GBLUPs and iidBLUPs as validation data.</strong></p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>accMeans <span class="sc">%&gt;%</span> </span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(ValidationData,Accuracy) <span class="sc">%&gt;%</span> </span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>GBLUPs<span class="sc">-</span>iidBLUPs) <span class="sc">%$%</span> <span class="fu">summary</span>(diffAcc)</span></code></pre></div>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
-0.02019  0.14109  0.17690  0.18111  0.22389  0.38530 </code></pre>
<p>Prediction accuracy for family means were nearly uniformly higher using GBLUPs compared to iidBLUPs (median 0.18 higher).</p>
<p><strong>Compute Spearman rank correlations on a per Trait, per prediction type (BV vs TGV) basis.</strong></p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>accMeans <span class="sc">%&gt;%</span> </span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model)) <span class="sc">%&gt;%</span> </span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(ValidationData,Accuracy) <span class="sc">%&gt;%</span> </span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait,predOf) <span class="sc">%&gt;%</span> </span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corAcc=</span><span class="fu">cor</span>(GBLUPs,iidBLUPs,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>)) <span class="sc">%$%</span> <span class="fu">summary</span>(corAcc) <span class="co"># arrange(desc(corAcc))</span></span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.5469  0.6921  0.7512  0.7344  0.7852  0.8438 </code></pre>
<p>The Spearman rank correlation between prediction accuracies based on iidBLUPs and GBLUPs was high (median 0.75, range 0.55-0.84).</p>
<p><strong>Summarize accuracy differences between GBLUPs and iidBLUPs as validation data.</strong></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>accVars <span class="sc">%&gt;%</span> </span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(ValidationData,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>GBLUPs<span class="sc">-</span>iidBLUPs) <span class="sc">%$%</span> <span class="fu">summary</span>(diffAcc)</span></code></pre></div>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
-0.38849 -0.00209  0.07280  0.06568  0.13568  0.43723 </code></pre>
<p>Similar to the means, accuracy using GBLUP-validation-data appeared mostly higher compared to iidBLUPs (median difference GBLUPs-iidBLUPs = 0.07, interquartile range -0.002-0.14).</p>
<p><strong>Compute Spearman rank correlations on a per Trait, per prediction type (BV vs TGV) basis.</strong></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>accVars <span class="sc">%&gt;%</span> </span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(ValidationData,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Component,TraitType,Trait1,Trait2,predOf) <span class="sc">%&gt;%</span> </span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corAcc=</span><span class="fu">cor</span>(GBLUPs,iidBLUPs,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>)) <span class="sc">%$%</span> <span class="fu">summary</span>(corAcc) <span class="co"># arrange(desc(corAcc))</span></span></code></pre></div>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.04154 0.33269 0.52385 0.50785 0.66346 0.89154 </code></pre>
<p>The Spearman rank correlations of iidBLUP and GBLUP-validation-based accuracies was positive for family (co)variances, but smaller compared to family means (mean correlation 0.5, range 0.04-0.89).</p>
<p>Supplementary plots comparing validation-data accuracies for means and (co)variances were inspected (<strong>Figure S??-S??</strong> <a href="SupplementaryFigures.html">Figure ???</a>). Based on this, we conclude that we would reach similar though more muted conclusions about which trait variances and trait-trait covariances are best or worst predicted, if restricted to iidBLUPs for validation data.</p>
</div>
<div id="accuracy-considering-only-families-with-10-members" class="section level2">
<h2>Accuracy considering only families with 10+ members?</h2>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>obsVSpredVars<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;obsVSpredVars.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>,VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData,<span class="sc">-</span>VarMethod)</span></code></pre></div>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>obsVSpredVars <span class="sc">%&gt;%</span> </span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sireID,damID,FamSize) <span class="sc">%&gt;%</span> </span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FamSize<span class="sc">&gt;=</span><span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span></code></pre></div>
<pre><code>[1] 118</code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>obsVSpredVars <span class="sc">%&gt;%</span> </span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sireID,damID,FamSize) <span class="sc">%&gt;%</span> </span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FamSize<span class="sc">&gt;=</span><span class="dv">20</span>) <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span></code></pre></div>
<pre><code>[1] 22</code></pre>
<p>In our primary analysis, we computed (co)variance prediction accuracies with weighted correlations, considering any family with more than one member. We also considered a more conservative alternative approach of including only families with $$10 (n=112) or $$20 (n=22).</p>
<p>Family Size &gt;= 20</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variances</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy<span class="ot">&lt;-</span>obsVSpredVars <span class="sc">%&gt;%</span> </span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(.) <span class="sc">%&gt;%</span> </span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FamSize<span class="sc">&gt;=</span><span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FamSize,<span class="sc">-</span>Nobs) <span class="sc">%&gt;%</span> </span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nest</span>(<span class="at">predVSobs=</span><span class="fu">c</span>(sireID,damID,predVar,obsVar,CorrWeight)) <span class="sc">%&gt;%</span> </span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FamSize10plus=</span><span class="fu">map_dbl</span>(predVSobs,<span class="sc">~</span>psych<span class="sc">::</span><span class="fu">cor.wt</span>(.[,<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>],<span class="at">w =</span> .<span class="sc">$</span>CorrWeight) <span class="sc">%$%</span> r[<span class="dv">1</span>,<span class="dv">2</span>])) <span class="sc">%&gt;%</span> </span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>predVSobs) <span class="sc">%&gt;%</span> </span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS11&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>,ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>, <span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model)) <span class="sc">%&gt;%</span> </span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData,<span class="sc">-</span>VarMethod) <span class="sc">%&gt;%</span> </span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">AllFams=</span>AccuracyWtCor)) <span class="sc">%&gt;%</span> </span>
<span id="cb108-14"><a href="#cb108-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">accDiff=</span>FamSize10plus<span class="sc">-</span>AllFams)</span>
<span id="cb108-15"><a href="#cb108-15" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy<span class="sc">$</span>accDiff <span class="sc">%&gt;%</span> summary</span></code></pre></div>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
-1.312734 -0.226769 -0.014017  0.003398  0.230687  1.088088 </code></pre>
<p>Family Size &gt;= 20</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy <span class="sc">%&gt;%</span> </span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corAllVSBig=</span><span class="fu">cor</span>(FamSize10plus,AllFams,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>,<span class="at">use=</span><span class="st">&#39;pairwise.complete.obs&#39;</span>))</span></code></pre></div>
<pre><code># A tibble: 1 x 1
  corAllVSBig
        &lt;dbl&gt;
1       0.388</code></pre>
<p>Family Size &gt;= 10</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variances</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy<span class="ot">&lt;-</span>obsVSpredVars <span class="sc">%&gt;%</span> </span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(.) <span class="sc">%&gt;%</span> </span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(FamSize<span class="sc">&gt;=</span><span class="dv">10</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>FamSize,<span class="sc">-</span>Nobs) <span class="sc">%&gt;%</span> </span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nest</span>(<span class="at">predVSobs=</span><span class="fu">c</span>(sireID,damID,predVar,obsVar,CorrWeight)) <span class="sc">%&gt;%</span> </span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FamSize10plus=</span><span class="fu">map_dbl</span>(predVSobs,<span class="sc">~</span>psych<span class="sc">::</span><span class="fu">cor.wt</span>(.[,<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>],<span class="at">w =</span> .<span class="sc">$</span>CorrWeight) <span class="sc">%$%</span> r[<span class="dv">1</span>,<span class="dv">2</span>])) <span class="sc">%&gt;%</span> </span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>predVSobs) <span class="sc">%&gt;%</span> </span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS11&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>,ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>, <span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model)) <span class="sc">%&gt;%</span> </span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>Model,<span class="sc">-</span>ValidationData,<span class="sc">-</span>VarMethod) <span class="sc">%&gt;%</span> </span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a>              <span class="fu">select</span>(<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a>              <span class="fu">rename</span>(<span class="at">AllFams=</span>AccuracyWtCor)) <span class="sc">%&gt;%</span> </span>
<span id="cb112-14"><a href="#cb112-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">accDiff=</span>FamSize10plus<span class="sc">-</span>AllFams)</span>
<span id="cb112-15"><a href="#cb112-15" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy<span class="sc">$</span>accDiff <span class="sc">%&gt;%</span> summary</span></code></pre></div>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
-0.33655 -0.05449  0.01336  0.01144  0.07475  0.31545 </code></pre>
<p>Family Size &gt;= 10</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy <span class="sc">%&gt;%</span> </span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corAllVSBig=</span><span class="fu">cor</span>(FamSize10plus,AllFams,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>,<span class="at">use=</span><span class="st">&#39;pairwise.complete.obs&#39;</span>))</span></code></pre></div>
<pre><code># A tibble: 1 x 1
  corAllVSBig
        &lt;dbl&gt;
1       0.893</code></pre>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>compareAllFamsToBigFamsAccuracy <span class="sc">%&gt;%</span> </span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait1,Trait2) <span class="sc">%&gt;%</span> </span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">medianDiffAccSize10PlusVsAllFams=</span><span class="fu">round</span>(<span class="fu">median</span>(FamSize10plus<span class="sc">-</span>AllFams),<span class="dv">3</span>)) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(medianDiffAccSize10PlusVsAllFams))</span></code></pre></div>
<pre><code># A tibble: 12 x 3
# Groups:   Trait1 [6]
   Trait1    Trait2    medianDiffAccSize10PlusVsAllFams
   &lt;chr&gt;     &lt;chr&gt;                                &lt;dbl&gt;
 1 DM        TCHART                               0.129
 2 logFYLD   MCMDS                                0.08 
 3 DM        logFYLD                              0.045
 4 logFYLD   logFYLD                              0.039
 5 DM        DM                                   0.019
 6 MCMDS     TCHART                               0.017
 7 TCHART    TCHART                               0.008
 8 MCMDS     MCMDS                                0.004
 9 biofortSI biofortSI                           -0.002
10 stdSI     stdSI                               -0.028
11 logFYLD   TCHART                              -0.06 
12 DM        MCMDS                               -0.062</code></pre>
<p>The Spearman rank correlation between accuracy estimates when all vs. only families with more than 10 members was 0.89. There should therefore be good concordance with our primary conclusions, depending on the family size threshold we impose.</p>
<p>The median difference in accuracy (“threshold size families” minus “all families”) was 0.01. Considering only size 10 or greater families noticeably improved prediction accuracy for several trait variances and especially for two covariances (DM-TCHART and logFYLD-MCMDS) (<strong>Figures S???</strong>).</p>
</div>
<div id="comparing-pmv-and-vpm-predictions" class="section level2">
<h2>Comparing PMV and VPM predictions</h2>
<p>Comparing posterior mean variance (PMV) to variance of posterior mean (VPM) predictions:</p>
<p>Variances and covariances estimates and cross-validation predictions were made with the PMV method, which is computationally more intensive than VPM.</p>
<p><strong>Overall Spearman rank correlation between VPM and PMV variance component estimates.</strong></p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S15: Variance estimates for genetic groups</span></span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>varcomps<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS15&quot;</span>)</span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>varcomps <span class="sc">%&gt;%</span> </span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Method<span class="sc">==</span><span class="st">&quot;M2&quot;</span>,<span class="fu">grepl</span>(<span class="st">&quot;DirDomAD&quot;</span>,Model)) <span class="sc">%&gt;%</span> </span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>propDom,<span class="sc">-</span>Model,<span class="sc">-</span>Method) <span class="sc">%&gt;%</span> </span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(VarA,VarD), <span class="at">names_to =</span> <span class="st">&quot;VarComp&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;VarEst&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(VarEst)) <span class="sc">%&gt;%</span> </span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(VarMethod,VarEst) <span class="sc">%&gt;%</span> </span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corPMV_VPM=</span><span class="fu">cor</span>(PMV,VPM, <span class="at">use =</span> <span class="st">&#39;pairwise.complete.obs&#39;</span>,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>))</span></code></pre></div>
<pre><code># A tibble: 1 x 1
  corPMV_VPM
       &lt;dbl&gt;
1      0.983</code></pre>
<p>We observed that population variance estimates based on PMV were consistently larger than VPM, but the correlation of those estimates is 0.98 (**Figure __**).</p>
<p><strong>Compute the correlation between PMV and VPM predictions.</strong></p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>obsVSpredVars<span class="ot">&lt;-</span><span class="fu">readRDS</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;obsVSpredVars.rds&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ValidationData,<span class="sc">-</span>Model)</span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>obsVSpredVars <span class="sc">%&gt;%</span> </span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>obsVar,<span class="sc">-</span>FamSize,<span class="sc">-</span>Nobs,<span class="sc">-</span>CorrWeight) <span class="sc">%&gt;%</span> </span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(VarMethod,predVar) <span class="sc">%&gt;%</span> </span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#group_by(predOf) %&gt;% </span></span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corPMV_VPM=</span><span class="fu">cor</span>(PMV,VPM, <span class="at">use =</span> <span class="st">&#39;pairwise.complete.obs&#39;</span>,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>))</span></code></pre></div>
<pre><code># A tibble: 1 x 1
  corPMV_VPM
       &lt;dbl&gt;
1      0.995</code></pre>
<p>Using the predictions from the cross-validation results, we further observed that the PMV predictions were consistently larger and most notably that the correlation between PMV and VPM was very high (0.995). Some VPM prediction accuracies actually appear better than PMV predictions (<strong>Figure S???</strong>).</p>
<p>The critical point is that VPM and PMV predictions should have very similar rankings. In our primary analysis, we focus on the PMV results with the only exception being the exploratory predictions where we saved time/computation and used the VPM. If implementing mate selections via the usefulness criteria, choosing the VPM method would mostly have the consequence of shrinking the influence on selection decisions towards the mean.</p>
</div>
<div id="compare-directional-dominance-to-classic-model" class="section level2">
<h2>Compare directional dominance to “classic” model</h2>
<p>Our focus in this article was not in finding the optimal or most accurate prediction model for obtaining marker effects. However, genome-wide estimates of directional dominance have not previously been made in cassava. For this reason, we make some brief comparison to the standard or “classic” additive-dominance prediction model, where dominance effects are centered on zero.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr);</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Table S10: Accuracies predicting the mean</span></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>accMeans<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS10&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ValidationData) <span class="sc">%&gt;%</span> </span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Model=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),<span class="st">&quot;DirDom&quot;</span>,<span class="st">&quot;Classic&quot;</span>))</span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Table S11: Accuracies predicting the variances</span></span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>accVars<span class="ot">&lt;-</span>readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTables.xlsx&quot;</span>),<span class="at">sheet =</span> <span class="st">&quot;TableS11&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Component=</span><span class="fu">ifelse</span>(Trait1<span class="sc">==</span>Trait2,<span class="st">&quot;Variance&quot;</span>,<span class="st">&quot;Covariance&quot;</span>),</span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">TraitType=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;SI&quot;</span>,Trait1),<span class="st">&quot;SI&quot;</span>,<span class="st">&quot;ComponentTrait&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ValidationData<span class="sc">==</span><span class="st">&quot;GBLUPs&quot;</span>,VarMethod<span class="sc">==</span><span class="st">&quot;PMV&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>ValidationData,<span class="sc">-</span>VarMethod,<span class="sc">-</span>AccuracyCor) <span class="sc">%&gt;%</span> </span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Model=</span><span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;DirDom&quot;</span>,Model),<span class="st">&quot;DirDom&quot;</span>,<span class="st">&quot;Classic&quot;</span>))</span></code></pre></div>
<p><strong>Compute rank correlations overall (DirDom vs Classic): Means</strong></p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>accMeans <span class="sc">%&gt;%</span> </span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Model,Accuracy) <span class="sc">%&gt;%</span>   </span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corMeanAcc=</span><span class="fu">cor</span>(DirDom,Classic,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>))</span></code></pre></div>
<pre><code># A tibble: 1 x 1
  corMeanAcc
       &lt;dbl&gt;
1      0.984</code></pre>
<p><strong>Compute rank correlations overall (DirDom vs Classic): Variances</strong></p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>accVars <span class="sc">%&gt;%</span> </span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Model,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corVarAcc=</span><span class="fu">cor</span>(DirDom,Classic,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>))</span></code></pre></div>
<pre><code># A tibble: 1 x 1
  corVarAcc
      &lt;dbl&gt;
1     0.938</code></pre>
<p>Overall, the ranking of models and predictions between the two models were similar, as indicated by a rank correlation between model accuracy estimates of 0.98 for family means and 0.94 for variances and covariances.</p>
<p><strong>Summarize accuracy differences between DirDom and Classic model: Means</strong></p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a>accMeans <span class="sc">%&gt;%</span> </span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Model,Accuracy) <span class="sc">%&gt;%</span> </span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>DirDom<span class="sc">-</span>Classic) <span class="sc">%$%</span> <span class="fu">summary</span>(diffAcc)</span></code></pre></div>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
-0.123522  0.002375  0.014843  0.024996  0.040209  0.233371 </code></pre>
<p><strong>Summarize accuracy differences between DirDom and Classic model: Variances</strong></p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a>accVars <span class="sc">%&gt;%</span> </span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Model,AccuracyWtCor) <span class="sc">%&gt;%</span> </span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">diffAcc=</span>DirDom<span class="sc">-</span>Classic) <span class="sc">%$%</span> <span class="fu">summary</span>(diffAcc)</span></code></pre></div>
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
-0.337929 -0.024895 -0.003922 -0.008563  0.013797  0.148698 </code></pre>
<p>Three-quarters of family-mean and almost half of (co)variance accuracy estimates were higher using the directional dominance model.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># **Compute rank correlations of models - cor(DirDom,Classic) - by Trait, predOf - Means**</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="co"># accMeans %&gt;% </span></span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   spread(Model,Accuracy) %&gt;%   </span></span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(Trait,predOf) %&gt;% </span></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarize(corAcc=cor(DirDom,Classic,method = &#39;spearman&#39;)) %&gt;% spread(predOf,corAcc) # %$% summary(corAcc) # arrange(desc(corAcc))</span></span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a><span class="co"># **Compute rank correlations of models - cor(DirDom,Classic) - by Trait, predOf - Variances**</span></span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  accVars %&gt;% </span></span>
<span id="cb131-10"><a href="#cb131-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   spread(Model,AccuracyWtCor) %&gt;% </span></span>
<span id="cb131-11"><a href="#cb131-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(Component,TraitType,Trait1,Trait2,predOf) %&gt;% </span></span>
<span id="cb131-12"><a href="#cb131-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarize(corAcc=round(cor(DirDom,Classic,method = &#39;spearman&#39;),3),</span></span>
<span id="cb131-13"><a href="#cb131-13" aria-hidden="true" tabindex="-1"></a><span class="co">#             medianDiffAcc=round(median(DirDom-Classic),3)) %&gt;% arrange(corAcc) %&gt;% rmarkdown::paged_table()</span></span></code></pre></div>
<p>The most notably improved predictions were for the family-mean logFYLD TGV (<strong>Figure S?? and S??</strong>).</p>
<p><strong>Overall correlations between all Classic and Directional Dominance Predictions of Untested Crosses</strong></p>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr); </span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses<span class="ot">&lt;-</span><span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;manuscript&quot;</span>,<span class="st">&quot;SupplementaryTable18.csv&quot;</span>),<span class="at">stringsAsFactors =</span> F) </span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a><span class="co"># predUntestedCrosses %&gt;% </span></span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   spread(Model,Pred) %&gt;% </span></span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(Trait,PredOf,Component) %&gt;% </span></span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarize(corModels=round(cor(ClassicAD,DirDom,method = &#39;spearman&#39;),2)) %&gt;% </span></span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   spread(Component,corModels) %&gt;% </span></span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   arrange(Trait,PredOf) %&gt;% </span></span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   rmarkdown::paged_table()</span></span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a>predUntestedCrosses <span class="sc">%&gt;%</span> </span>
<span id="cb132-11"><a href="#cb132-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(Model,Pred) <span class="sc">%&gt;%</span> </span>
<span id="cb132-12"><a href="#cb132-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">corModels=</span><span class="fu">round</span>(<span class="fu">cor</span>(ClassicAD,DirDom,<span class="at">method =</span> <span class="st">&#39;spearman&#39;</span>),<span class="dv">2</span>))</span></code></pre></div>
<pre><code>  corModels
1      0.98</code></pre>
<p>There was also an overall rank correlation of 0.98 between models in the prediction of untested crosses.</p>
</div>
<div id="convergence" class="section level2">
<h2>Convergence</h2>
<p>Visual check of the posterior distribution and trace of error variance estimates, to check convergence.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse); <span class="fu">library</span>(magrittr); <span class="fu">library</span>(patchwork); <span class="fu">library</span>(ragg)</span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>cv_brrIterHist<span class="ot">&lt;-</span><span class="fu">list.files</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output/mtMarkerEffects&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span>(<span class="st">&quot;.dat&quot;</span>,.,<span class="at">value =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span>(<span class="st">&quot;Repeat&quot;</span>,.,<span class="at">value =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">Filename=</span>.) <span class="sc">%&gt;%</span> </span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> Filename,</span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;prefix&quot;</span>,<span class="st">&quot;rep&quot;</span>,<span class="st">&quot;fold&quot;</span>,<span class="st">&quot;dataset&quot;</span>,<span class="st">&quot;Model&quot;</span>,<span class="st">&quot;extra&quot;</span>),</span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>,<span class="at">remove =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb134-9"><a href="#cb134-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Dataset=</span><span class="fu">paste</span>(rep,fold,dataset,Model,<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb134-10"><a href="#cb134-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Dataset,Model,Filename)</span>
<span id="cb134-11"><a href="#cb134-11" aria-hidden="true" tabindex="-1"></a>gengroup_brrIterHist<span class="ot">&lt;-</span><span class="fu">list.files</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output/mtMarkerEffects&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb134-12"><a href="#cb134-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span>(<span class="st">&quot;.dat&quot;</span>,.,<span class="at">value =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb134-13"><a href="#cb134-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span>(<span class="st">&quot;Repeat&quot;</span>,.,<span class="at">value =</span> T, <span class="at">invert =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb134-14"><a href="#cb134-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">Filename=</span>.) <span class="sc">%&gt;%</span> </span>
<span id="cb134-15"><a href="#cb134-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> Filename,</span>
<span id="cb134-16"><a href="#cb134-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;prefix&quot;</span>,<span class="st">&quot;Dataset&quot;</span>,<span class="st">&quot;Model&quot;</span>,<span class="st">&quot;extra&quot;</span>),</span>
<span id="cb134-17"><a href="#cb134-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>,<span class="at">remove =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb134-18"><a href="#cb134-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Dataset=</span><span class="fu">paste0</span>(Dataset,<span class="st">&quot;_&quot;</span>,Model)) <span class="sc">%&gt;%</span> </span>
<span id="cb134-19"><a href="#cb134-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Dataset,Model,Filename)</span>
<span id="cb134-20"><a href="#cb134-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb134-21"><a href="#cb134-21" aria-hidden="true" tabindex="-1"></a>brrIterHist<span class="ot">&lt;-</span><span class="fu">bind_rows</span>(gengroup_brrIterHist,cv_brrIterHist) <span class="sc">%&gt;%</span> </span>
<span id="cb134-22"><a href="#cb134-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">iterHist=</span><span class="fu">map</span>(Filename,<span class="sc">~</span><span class="fu">read.table</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;output/mtMarkerEffects/&quot;</span>),.),</span>
<span id="cb134-23"><a href="#cb134-23" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">header =</span> F, <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span> </span>
<span id="cb134-24"><a href="#cb134-24" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">mutate</span>(<span class="at">Iteration=</span><span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb134-25"><a href="#cb134-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb134-26"><a href="#cb134-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="fu">c</span>(V1<span class="sc">:</span>V10),<span class="at">values_to =</span> <span class="st">&quot;varR&quot;</span>,<span class="at">names_to =</span> <span class="st">&quot;varCovarParam&quot;</span>)</span>
<span id="cb134-27"><a href="#cb134-27" aria-hidden="true" tabindex="-1"></a><span class="co"># brrIterHist$iterHist[[1]] %&gt;% slice()</span></span>
<span id="cb134-28"><a href="#cb134-28" aria-hidden="true" tabindex="-1"></a><span class="co"># brrIterHist %&gt;% slice(1) %&gt;% unnest(iterHist)</span></span></code></pre></div>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>colors<span class="ot">&lt;-</span>viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">4</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>brrIterHist <span class="sc">%&gt;%</span> </span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.,<span class="fu">aes</span>(<span class="at">x=</span>Iteration,<span class="at">y=</span>varR,<span class="at">color=</span>Model,<span class="at">group=</span>Dataset)) <span class="sc">+</span> </span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">alpha=</span><span class="fl">0.7</span>,<span class="at">size=</span><span class="fl">0.8</span>) <span class="sc">+</span> </span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(varCovarParam<span class="sc">~</span>., <span class="at">scales =</span> <span class="st">&#39;free_y&#39;</span>) <span class="sc">+</span> </span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a> <span class="co"># facet_wrap(~varCovarParam,scales = &#39;free_y&#39;,ncol=1) + </span></span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span>  </span>
<span id="cb135-10"><a href="#cb135-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>),</span>
<span id="cb135-11"><a href="#cb135-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co">#element_text(face=&#39;bold&#39;,color = &#39;black&#39;),</span></span>
<span id="cb135-12"><a href="#cb135-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>,<span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb135-13"><a href="#cb135-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb135-14"><a href="#cb135-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb135-15"><a href="#cb135-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb135-16"><a href="#cb135-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&#39;bold&#39;</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb135-17"><a href="#cb135-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>),</span>
<span id="cb135-18"><a href="#cb135-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&#39;bottom&#39;</span>,</span>
<span id="cb135-19"><a href="#cb135-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb135-20"><a href="#cb135-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb135-21"><a href="#cb135-21" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb135-22"><a href="#cb135-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colors) <span class="sc">+</span> </span>
<span id="cb135-23"><a href="#cb135-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Thinned Iteration&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Error (Co)variance&quot;</span>)</span></code></pre></div>
<p><img src="figure/Results.Rmd/unnamed-chunk-79-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a>brrIterHist <span class="sc">%&gt;%</span> </span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Dataset,Model,varCovarParam,Iteration) <span class="sc">%&gt;%</span> </span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Dataset,Model,varCovarParam) <span class="sc">%&gt;%</span> </span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice</span>(<span class="dv">1001</span><span class="sc">:</span><span class="dv">6000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb136-5"><a href="#cb136-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb136-6"><a href="#cb136-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Dataset,Model,varCovarParam,Iteration) <span class="ot">-&gt;</span> x</span></code></pre></div>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>colors<span class="ot">&lt;-</span>viridis<span class="sc">::</span><span class="fu">viridis</span>(<span class="dv">10</span>)</span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>x <span class="sc">%&gt;%</span>   </span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(.,<span class="fu">aes</span>(<span class="at">x=</span>varR,<span class="at">fill=</span>Model)) <span class="sc">+</span> </span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha=</span><span class="fl">0.7</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>) <span class="sc">+</span> </span>
<span id="cb137-5"><a href="#cb137-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb137-6"><a href="#cb137-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  facet_grid(varCovarParam~Model, scales = &#39;free&#39;) + </span></span>
<span id="cb137-7"><a href="#cb137-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>varCovarParam,<span class="at">scales =</span> <span class="st">&#39;free&#39;</span>) <span class="sc">+</span> </span>
<span id="cb137-8"><a href="#cb137-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span>  </span>
<span id="cb137-9"><a href="#cb137-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>),</span>
<span id="cb137-10"><a href="#cb137-10" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(), <span class="co">#element_text(face=&#39;bold&#39;,color = &#39;black&#39;),</span></span>
<span id="cb137-11"><a href="#cb137-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>,<span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb137-12"><a href="#cb137-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb137-13"><a href="#cb137-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&#39;bold&#39;</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>, <span class="at">angle=</span><span class="dv">30</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb137-14"><a href="#cb137-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&#39;bold&#39;</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb137-15"><a href="#cb137-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&#39;bold&#39;</span>,<span class="at">color=</span><span class="st">&#39;black&#39;</span>),</span>
<span id="cb137-16"><a href="#cb137-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&#39;bold&#39;</span>),</span>
<span id="cb137-17"><a href="#cb137-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&#39;bottom&#39;</span>,</span>
<span id="cb137-18"><a href="#cb137-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb137-19"><a href="#cb137-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb137-20"><a href="#cb137-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb137-21"><a href="#cb137-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colors) <span class="sc">+</span> </span>
<span id="cb137-22"><a href="#cb137-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Distribution of Post burn-in Error (co)variance estimates&quot;</span>)</span></code></pre></div>
<p><img src="figure/Results.Rmd/unnamed-chunk-81-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.1.1 ragg_1.1.2      magrittr_2.0.1  forcats_0.5.1  
 [5] stringr_1.4.0   dplyr_1.0.5     purrr_0.3.4     readr_1.4.0    
 [9] tidyr_1.1.3     tibble_3.1.0    ggplot2_3.3.3   tidyverse_1.3.0
[13] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] viridis_0.5.1      httr_1.4.2         sass_0.3.1         jsonlite_1.7.2    
 [5] viridisLite_0.3.0  tmvnsim_1.0-2      here_1.0.1         modelr_0.1.8      
 [9] bslib_0.2.4        assertthat_0.2.1   highr_0.8          cellranger_1.1.0  
[13] yaml_2.2.1         corrplot_0.84      lattice_0.20-41    pillar_1.5.1      
[17] backports_1.2.1    glue_1.4.2         digest_0.6.27      promises_1.2.0.1  
[21] rvest_1.0.0        colorspace_2.0-0   htmltools_0.5.1.1  httpuv_1.5.5      
[25] psych_2.0.12       pkgconfig_2.0.3    broom_0.7.5        haven_2.3.1       
[29] scales_1.1.1       whisker_0.4        later_1.1.0.1      git2r_0.28.0      
[33] generics_0.1.0     farver_2.1.0       ellipsis_0.3.1     withr_2.4.1       
[37] cli_2.3.1          mnormt_2.0.2       crayon_1.4.1       readxl_1.3.1      
[41] evaluate_0.14      fs_1.5.0           fansi_0.4.2        nlme_3.1-152      
[45] xml2_1.3.2         textshaping_0.3.3  tools_4.0.3        hms_1.0.0         
[49] lifecycle_1.0.0    munsell_0.5.0      reprex_1.0.0       compiler_4.0.3    
[53] jquerylib_0.1.3    systemfonts_1.0.1  gridGraphics_0.5-1 rlang_0.4.10      
[57] grid_4.0.3         rstudioapi_0.13    labeling_0.4.2     rmarkdown_2.7     
[61] gtable_0.3.0       DBI_1.1.1          R6_2.5.0           gridExtra_2.3     
[65] lubridate_1.7.10   knitr_1.31         utf8_1.2.1         rprojroot_2.0.2   
[69] stringi_1.5.3      parallel_4.0.3     Rcpp_1.0.6         vctrs_0.3.6       
[73] dbplyr_2.1.0       tidyselect_1.1.0   xfun_0.22         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
